Software/Development/SemesterProject/App/dutyCycle.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * dutyCycle.c
+ *
+ *  Created on: 12 mar. 2019
+ *      Author: Aitor Teran
+ */
+//
+//###########################################################################
+
+//###########################################################################
+
+
+
+//
+// Includes
+//
+#include "Include/dutyCycle.h"
+#include "../System/Include/PWMConfig.h"
+
+
+//
+// Quasi-global variables definition
+//
+static struct dutyCycleTag
+{
+    DutyCycle legA;
+    DutyCycle legB;
+    DutyCycle legC;
+}dutyCycle;
+
+
+/* CONSIDER REVISION to reduce computing load.
+ * Function for internal use.
+ * map function performs a linear regression.
+ * Inputs are a value with its corresponding lower and upper limits
+ * as well as the lower and upper limits of the variable to return.
+ */
+Uint16 mapDuty(float x, float x_min, float x_max, Uint16 y_min, Uint16 y_max)
+{
+    return y_min + (Uint16)((float)(y_max-y_min)*(x-x_min)/(x_max-x_min));
+}
+
+/*
+ * Function for internal use.
+ * The duty cycle is checked in case it lies outside of boundaries.
+ * Later the struct containing the duty information gets updated.
+ */
+void fillDutyStruct(float duty, int leg)
+{
+
+    //Counter compare module uses the time counter variable to compare in order to set the
+    //correct PWM duty cycle, for this, it is important to know how many counts there are in a cycle.
+
+    //In the PWMConfig.c file, the period of the duty cycle is set by assigning the number of clk of a
+    //a period, this is done in the command 'EPwm1Regs.TBPRD = INTERNAL_FREQ/(2*SW_FREQ);'
+    //Since the configured carrier is a symmetrical triangular, the real period is actually twice as
+    //much but the counter will reach maximum that value.
+
+    //duty cycle float variable must then be converted into a Uint16 with correspondent
+    //input 0.00 -> output 0 and input 100.00 -> output INTERNAL_FREQ/(2*SW_FREQ).
+
+    //Error check.
```

Comment:
We haven't really talked about this, but I think the best pattern is to perform the analysis of the data upstream to whenever duty value is generated. Otherwise every function should have this kind of error check for every argument.

Reply:
Ok, i agree, then dutyCycle will not include any kind of error check. I take it out from here.